sweetest/src/main/java/com/mysugr/sweetest/framework/coroutine/CoroutinesTestContext.kt

```diff
@@ -0,0 +1,23 @@
+package com.mysugr.sweetest.framework.coroutine
+
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.newSingleThreadContext
+import kotlin.coroutines.CoroutineContext
+
+class CoroutinesTestContext {
+    private val name = CoroutineName("testCoroutine${instanceCounter++}")
+    private val supervisorJob = SupervisorJob()
+    val coroutineContext: CoroutineContext
+        get() = coroutineDispatcher + supervisorJob + name
+
+    suspend fun testFinished() {
```

Comment:
i think it's better to use a name that actually says what the function does (especially if it's something that simple) and/or use verbs for function names, e.g. `cancel`, but i'm not sure if it should also have an explicit `Blocking` suffix?